syntax = "proto3";

package wukong.proto;

message Message {
  enum MessageType {
    FUNCTION = 0;   //调用函数
  }
  uint32 id = 1;
  MessageType type = 2;

  string application = 3;
  string function = 4;

  bool isAsync = 5;

  bytes inputData = 6;
  bytes outputData = 7;
  string resultKey = 8;

  int64 timestamp = 9;
  int64 finishTimestamp = 10;

}

message Function{
  string user = 1;
  string application = 2;
  string functionName = 3;
  uint32 concurrency = 4;
  uint32 memory = 5;
  uint32 cpu = 6;
  string storageKey = 7;
}

message Invoker{
  string invokerID = 1;
  string IP = 2;
  uint32 port = 3;
  uint32 memory = 4;
  uint32 cpu = 5;
  uint64 registerTime = 6;
}

message ReplyRegisterInvoker{
  bool success = 1;
  string invokerID = 2;
  string msg = 3;
}